tests/conftest.py

```diff
@@ -10,61 +10,188 @@ def isolation_setup(fn_isolation):
 
 @pytest.fixture(scope="module")
 def token(ERC20, accounts):
-    yield ERC20.deploy("Yearn Token", "YFI", 18, {"from": accounts[0]})
+    return ERC20.deploy("Lido Token", "LFI", 18, {"from": accounts[0]})
+
+
+@pytest.fixture(scope="module")
+def voting(Voting, token, owner):
+    return Voting.deploy(token, {"from": owner})
+
+
+@pytest.fixture(scope="module")
+def delegate(Delegate, owner):
+    return Delegate.deploy({"from": owner})
+
+
+@pytest.fixture(scope="module")
+def voting_adapter(VotingAdapter, owner):
+    return VotingAdapter.deploy({"from": owner})
+
+
+@pytest.fixture(scope="module")
+def voting_adapter_for_update(VotingAdapter, owner):
+    return VotingAdapter.deploy({"from": owner})
 
 
 @pytest.fixture(scope="module")
 def start_time(chain):
-    yield chain.time() + 1000 + 86400 * 365
+    return chain.time() + 1000 + 86400 * 365
 
 
 @pytest.fixture(scope="module")
-def end_time(start_time):
-    yield int(start_time + 3 * YEAR)
+def end_time(start_time, duration):
+    return int(start_time + duration)
 
 
 @pytest.fixture(scope="module")
-def vesting_target(VestingEscrowSimple, accounts):
-    yield VestingEscrowSimple.deploy({"from": accounts[0]})
+def vesting_target_simple(VestingEscrow, owner, voting, delegate):
+    return VestingEscrow.deploy(voting, delegate, {"from": owner})
 
 
 @pytest.fixture(scope="module")
-def vesting_factory(VestingEscrowFactory, accounts, vesting_target):
-    yield VestingEscrowFactory.deploy(vesting_target, {"from": accounts[0]})
+def vesting_target_fully_revokable(
+    VestingEscrowFullyRevokable, owner, voting, delegate
+):
+    return VestingEscrowFullyRevokable.deploy(voting, delegate, {"from": owner})
 
 
 @pytest.fixture(scope="module")
-def vesting(VestingEscrowSimple, accounts, vesting_factory, token, start_time):
-    token._mint_for_testing(10 ** 20, {"from": accounts[0]})
-    token.approve(vesting_factory, 10 ** 20, {"from": accounts[0]})
+def vesting_factory(
+    VestingEscrowFactory,
+    owner,
+    vesting_target_simple,
+    vesting_target_fully_revokable,
+    voting_adapter,
+):
+    return VestingEscrowFactory.deploy(
+        vesting_target_simple,
+        vesting_target_fully_revokable,
+        voting_adapter,
+        {"from": owner},
+    )
+
+
+@pytest.fixture(
+    scope="module",
+    params=[pytest.param(0, id="simple"), pytest.param(1, id="fully_revocable")],
+)
+def deployed_vesting(
+    VestingEscrow,
+    VestingEscrowFullyRevokable,
+    recipient,
+    vesting_factory,
+    token,
+    start_time,
+    duration,
+    owner,
+    manager,
+    request,
+    balance,
+):
+    token._mint_for_testing(balance, {"from": owner})
+    token.approve(vesting_factory, balance, {"from": owner})
     tx = vesting_factory.deploy_vesting_contract(
         token,
-        accounts[1],
-        10 ** 20,
-        3 * YEAR,  # duration
+        balance,
+        recipient,
+        owner,
+        duration,  # duration
         start_time,
         0,  # cliff
```

Comment:
Why no cliff always?

Reply:
Nice catch! 💪
Added tests for claims before and after cliff